benchmarks/babelstream/src/babelStreamMainTest.cpp

```diff
@@ -312,136 +345,252 @@ void testKernels()
+            alpaka::wait(queue);
 
+            DataType const* sumPtr = std::data(bufHostSumPerBlock);
+            float const result = std::reduce(sumPtr, sumPtr + gridBlockExtent, 0.0f);
 
-    //
-    // Calculate and Display Benchmark Results
-    //
-    std::vector<double> bytesReadWriteMB = {
-        getDataThroughput<DataType>(2u, static_cast<unsigned>(arraySize)),
-        getDataThroughput<DataType>(2u, static_cast<unsigned>(arraySize)),
-        getDataThroughput<DataType>(2u, static_cast<unsigned>(arraySize)),
-        getDataThroughput<DataType>(3u, static_cast<unsigned>(arraySize)),
-        getDataThroughput<DataType>(3u, static_cast<unsigned>(arraySize)),
-        getDataThroughput<DataType>(2u, static_cast<unsigned>(arraySize)),
-    };
+            float const expectedSum = static_cast<float>(static_cast<DataType>(arraySize) * expectedA * expectedB);
 
-    // calculate the bandwidth as throughput per seconds
-    std::vector<double> bandwidthsPerKernel;
-    if(minExecTimesOfKernels.size() == kernelLabels.size())
+            //  Dot product should be identical to arraySize*valA*valB
+            //  Use a different equality check if floating point errors exceed precision of FuzzyEqual function
+            REQUIRE(FuzzyEqual((result - expectedSum) / expectedSum, 0.0f));
+
+            // Add workdiv to the list of workdivs to print later
+            metaData.setItem(BMInfoDataType::WorkDivDot, workDivDot);
+        }
+
+        //
+        // Calculate and Display Benchmark Results for Kernels
+        //
+        bytesReadWriteMB = {
+            getDataThroughput<DataType>(2u, static_cast<unsigned>(arraySize)), // init
+            getDataThroughput<DataType>(2u, static_cast<unsigned>(arraySize)), // copy
+            getDataThroughput<DataType>(2u, static_cast<unsigned>(arraySize)), // mul
+            getDataThroughput<DataType>(3u, static_cast<unsigned>(arraySize)), // add
+            getDataThroughput<DataType>(3u, static_cast<unsigned>(arraySize)), // triad
+            getDataThroughput<DataType>(2u, static_cast<unsigned>(arraySize)), // dot
+        };
+
+        // calculate the bandwidth as throughput per seconds
+
+        if(minExecTimesOfKernels.size() == kernelLabels.size())
+        {
+            for(size_t i = 0; i < minExecTimesOfKernels.size(); ++i)
+            {
+                bandwidthsPerKernel.push_back(calculateBandwidth(bytesReadWriteMB.at(i), minExecTimesOfKernels.at(i)));
+            }
+        }
+
+        metaData.setItem(BMInfoDataType::WorkDivInit, workDivInit);
+        metaData.setItem(BMInfoDataType::WorkDivCopy, workDivCopy);
+        metaData.setItem(BMInfoDataType::WorkDivAdd, workDivAdd);
+        metaData.setItem(BMInfoDataType::WorkDivMult, workDivMult);
+        metaData.setItem(BMInfoDataType::WorkDivTriad, workDivTriad);
+
+    } // for all 5 kernels and dot if accelerator is suitable
```

Comment:
I do not see the verification if all benchmarks will be executed. Please use the verification provided upstream this will ensure all is correct without doing a verification independent for all combinations.

https://github.com/UoB-HPC/BabelStream/blob/2f00dfb7f8b7cfe8c53d20d5c770bccbf8673440/src/main.cpp#L513

Reply:
The lines 419 to 440 verifies init and 4 kernels. Dot kernel is verified separately. I guess you mean reorganization of verification so that it will be cleaner like in the link?